uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -285,7 +291,14 @@ define([
             initAbsenceType();
             initStatus();
             initContact();
-            initDates();
+
+            //the promise will set the day types so that initial value is set properly
+            return initDates();
```

Comment:
If `initDates` needs to be used to respect the order of execution while the other functions (`initAbsenceType`, `initStatus`, etc) don't, then it's better to separate it for readability sake
```js
.then(function () {
  return $q.all([loadDayTypes(), loadStatuses()]);
})
.then(function () {
  return initDates();
})
.then(function () {
  initAbsenceType();
  initStatus();
  initContact();
})